Sage/carbon
# fix(popover-container): ensure clicks inside popover do not close it

Fix bug where portals inside a component did not automatically work with useClickAwayListener.

Refactor useClickAwayListener to no longer take an array of refs and return a stopPropagation handler which can be attached to the parent component - this automatically handles any React children no matter whether they are rendered as DOM children or not. This ensure that eg a Select inside a Popover Container will not cause the popover to close when a select option is clicked.

Remove any refs in parent components which are now unneeded.

fix #5322

### Proposed behaviour
Clicking inside a Popover Container should not close it



### Current behaviour
Clicking inside a Popover Container causes it to close (since version 109.1.3)



### Checklist




#### QA


### Additional context



### Testing instructions



The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/keen-sammet-zuw4wk?file=/src/App.js

Also test the case with a Select inside a Popover Container (mentioned towards the bottom of the original bug report on Github) - it should be possible to choose an option from the Select without the popover closing. This can be checked with the new Test story "with select" in Popover Container (and a Cypress test has been added to guard against regressions).
